python/lsst/analysis/tools/tasks/cpAnalysis.py

```diff
@@ -0,0 +1,361 @@
+):
+    data = cT.Input(
+        doc="Input data from cp_verify.",
+        name="verifyCalibStats",
+        storageClass="StructuredDataDict",
+        dimensions=("instrument",),
+    )
+    rawData = cT.Input(
+        doc="Raw values that should be collated earlier.",
+        name="verifyCalibDetStats",
+        storageClass="StructuredDataDict",
+        dimensions=(
+            "instrument",
+            "exposure",
+            "detector",
+        ),
+        deferLoad=True,
+        multiple=True,
+    )
+
+    camera = cT.PrerequisiteInput(
+        doc="Input camera to use for focal plane geometry.",
+        name="camera",
+        storageClass="Camera",
+        dimensions=("instrument",),
+        isCalibration=True,
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+
+class CpAnalysisConfig(AnalysisBaseConfig, pipelineConnections=CpAnalysisConnections):
+    def setDefaults(self):
+        super().setDefaults()
+
+
+class CpAnalysisCalibConnections(
+    AnalysisBaseConnections,
+    dimensions=("instrument",),
+    defaultTemplates={"outputName": "cpCalibTest"},
+):
+    data = cT.Input(
+        doc="Input data from cp_verify.",
+        name="verifyCalibStats",
+        storageClass="StructuredDataDict",
+        dimensions=("instrument",),
+    )
+    rawData = cT.Input(
+        doc="Raw values that should be collated earlier.",
+        name="verifyCalibDetStats",
+        storageClass="StructuredDataDict",
+        dimensions=(
+            "instrument",
+            "detector",
+        ),
+        deferLoad=True,
+        multiple=True,
+    )
+
+    camera = cT.PrerequisiteInput(
+        doc="Input camera to use for focal plane geometry.",
+        name="camera",
+        storageClass="Camera",
+        dimensions=("instrument",),
+        isCalibration=True,
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+
+class CpAnalysisCalibConfig(AnalysisBaseConfig, pipelineConnections=CpAnalysisCalibConnections):
+    def setDefaults(self):
+        super().setDefaults()
+
+
+class CpAnalysisTask(AnalysisPipelineTask):
+    ConfigClass = CpAnalysisConfig
+    _DefaultName = "cpAnalysisTask"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # Docs inherited from base class.
+        inputs = butlerQC.get(inputRefs)
+        dataId = butlerQC.quantum.dataId
+
+        data = self.loadData(inputs["rawData"])
+        # import pdb; pdb.set_trace()
+
+        band = None
+        if "physical_filter" in dataId.keys():
+            band = dataId["physical_filter"]
+        instrument = dataId["instrument"]
+        plotInfo = {
+            "run": inputs["rawData"][0].ref.run,
+            "tableName": f"{instrument}: bias",
+            "bands": [band],
+        }
+        outputs = self.run(data=data, plotInfo=plotInfo, camera=inputs["camera"])
+        butlerQC.put(outputs, outputRefs)
+
+    def loadData(self, rawData):
```

Comment:
You should be able to add `@abc.abstractmethod` above and have the class inherit from `abc.ABC`.